RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -0,0 +1,207 @@
+    std::pair<std::vector<T>,std::vector<T>> categorise(const std::vector<T>& pfMatchedJets, double pt1, double pt2, double Mjj)
+    {
+        std::pair<std::vector<T>,std::vector<T>> output;
+        unsigned int i1 = 0;
+        unsigned int i2 = 0;
+        double mjj = 0;
+        if (pfMatchedJets.size()>1){
+            for (unsigned int i = 0; i < pfMatchedJets.size()-1; i++){
+                
+                const T &  myJet1 = (pfMatchedJets)[i];
+                
+                for (unsigned int j = i+1; j < pfMatchedJets.size(); j++)
+                {
+                    const T &  myJet2 = (pfMatchedJets)[j];
+                    
+                    const double mjj_test = (myJet1.p4()+myJet2.p4()).M();
+                    
+                    if (mjj_test > mjj){
+                        
+                        mjj =mjj_test;
+                        i1 = i;
+                        i2 = j;
+                    }
+                }
+            }
+            
+            const T &  myJet1 = (pfMatchedJets)[i1];
+            const T &  myJet2 = (pfMatchedJets)[i2];
+            
+            if ((mjj > Mjj) && (myJet1.pt() >= pt1) && (myJet2.pt() > pt2) )
+            {
+                
+                output.first.push_back(myJet1);
+                output.first.push_back(myJet2);
+                
+            }
+            
+            if ((mjj > Mjj) && (myJet1.pt() < pt1) && (myJet1.pt() > pt2) && (myJet2.pt() > pt2))
+            {
+                
+                const T &  myJetTest = (pfMatchedJets)[0];
+                if (myJetTest.pt()>pt1){
+                    output.second.push_back(myJet1);
+                    output.second.push_back(myJet2);
+                    output.second.push_back(myJetTest);
+                    
+                }
+            }
+            
+        }
+        
+        return output;
+        
+    }
+    template< typename T>
+    L1TJetsMatching<T>::L1TJetsMatching(const edm::ParameterSet& iConfig):
+    jetSrc_    ( consumes<std::vector<T>>                     (iConfig.getParameter<InputTag>("JetSrc"      ) ) ),
+    jetTrigger_( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<InputTag>("L1JetTrigger") ) ),
+    pt1_Min   ( iConfig.getParameter<double>("pt1_Min")),
+    pt2_Min   ( iConfig.getParameter<double>("pt2_Min")),
+    mjj_Min   ( iConfig.getParameter<double>("mjj_Min"))
+    {
+        produces<std::vector<T>>("TwoJets");
+        produces<std::vector<T>>("ThreeJets");
+    }
+    template< typename T>
+    L1TJetsMatching<T>::~L1TJetsMatching(){ }
+    
+    template< typename T>
+    void L1TJetsMatching<T>::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+    {
+        
+        unique_ptr<std::vector<T>> pfMatchedJets(new std::vector<T>);
+        std::pair<std::vector<T>,std::vector<T>> output;
+        
+        double deltaR2    = 1.0;
+        double matchingR2 = 0.25;
```

Comment:
Ooops missed that!
I suggest to make matchingR  a configurable parameters of the class (iConfig.get and fillDescriptions)
and square it in the c'tor. deltaR2 is calculated later, so does not need to be configurable.